packages/react/dialog/src/Dialog.tsx

```diff
@@ -257,11 +258,15 @@ const DialogContentNonModal = React.forwardRef((props, forwardedRef) => {
         disableOutsidePointerEvents={false}
         onCloseAutoFocus={composeEventHandlers(props.onCloseAutoFocus, (event) => {
           event.preventDefault();
-          if (isEscapeKeyDownRef.current) context.triggerRef.current?.focus();
-        })}
-        onEscapeKeyDown={composeEventHandlers(props.onEscapeKeyDown, () => {
-          isEscapeKeyDownRef.current = true;
+          if (!isPointerDownOutsideRef.current) context.triggerRef.current?.focus();
         })}
+        onPointerDownOutside={composeEventHandlers(
+          props.onPointerDownOutside,
+          () => {
+            isPointerDownOutsideRef.current = true;
+          },
+          { checkForDefaultPrevented: false }
+        )}
```

Comment:
if you fancy being generous haha, nit 🙈 :

```suggestion
        onPointerDownOutside={composeEventHandlers(
          props.onPointerDownOutside,
          () => (isPointerDownOutsideRef.current = true),
          { checkForDefaultPrevented: false }
        )}
```